scene/resources/texture.cpp

```diff
@@ -1489,7 +1493,15 @@ void AtlasTexture::set_atlas(const Ref<Texture2D> &p_atlas) {
 	if (atlas == p_atlas) {
 		return;
 	}
+	// Support recursive AtlasTextures.
+	if (Ref<AtlasTexture>(atlas).is_valid()) {
+		atlas->disconnect(CoreStringNames::get_singleton()->changed, callable_mp(this, &AtlasTexture::_sub_atlas_texture_changed));
```

Comment:
You can likely connect `emit_changed` directly instead of a helper method (with `callable_mp(this, &Resource::emit_changed)`, might need casting `this`, not sure.

Reply:
Yeah:
```
scene/resources/tile_set.cpp
4029:   tad.alternatives[0]->connect("changed", callable_mp((Resource *)this, &TileSetAtlasSource::emit_changed));
```

Reply:
Thank you. Pushed this change for the time being. Will also try to update this PR's description to explain what goes on a bit more deeply.